modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java

```diff
@@ -0,0 +1,161 @@
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.core.api.el.ExpressionManager;
+import org.mule.runtime.module.extension.internal.loader.java.property.InjectableParameterInfo;
+import org.mule.runtime.module.extension.internal.runtime.ValueResolvingException;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ParameterValueResolver;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+
+/**
+ * Class in charge of resolving the values of acting parameter given the list of {@link InjectableParameterInfo} , the
+ * {@link ParameterizedModel} it is associated to and the {@link ParameterValueResolver} that corresponds to it.
+ *
+ * @since 4.4.0
+ */
+public class InjectableParameterResolver {
+
+  private static final Logger LOGGER = getLogger(InjectableParameterResolver.class);
+
+  private static final String EXPRESSION_PREFIX = "#[{";
+  private static final String EXPRESSION_SUFFIX = "}]";
+  private static final String EMPTY_EXPRESSION = EXPRESSION_PREFIX + EXPRESSION_SUFFIX;
+
+  private final Map<String, Object> resolvedParameterValues;
+  private final ExpressionManager expressionManager;
+  private final Map<String, InjectableParameterInfo> injectableParametersMap;
+
+  public InjectableParameterResolver(ParameterizedModel parameterizedModel,
+                                     ParameterValueResolver parameterValueResolver,
+                                     ExpressionManager expressionManager,
+                                     List<InjectableParameterInfo> injectableParameters) {
+    this.expressionManager = expressionManager;
+    this.injectableParametersMap = getInjectableParametersMap(injectableParameters);
+    this.resolvedParameterValues = getResolvedValues(parameterValueResolver, parameterizedModel);
+  }
+
+  /**
+   * Retrieves the value of a injectable parameter
+   *
+   * @param parameterName the name of the injectable parameter.
+   * @return the value of the injectable parameter.
+   */
+  public Object getInjectableParameterValue(String parameterName) {
+    Object parameterValue;
+    parameterValue = resolvedParameterValues.get(parameterName);
+    InjectableParameterInfo injectableParameterInfo = injectableParametersMap.get(parameterName);
+
+    if (parameterValue != null) {
+      try {
+        parameterValue = expressionManager
+            .evaluate("#[payload]",
+                      DataType.fromType(getClassFromType(injectableParameterInfo.getType())),
+                      BindingContext.builder()
+                          .addBinding("payload", new TypedValue(parameterValue, DataType.fromObject(parameterValue))).build())
+            .getValue();
+      } catch (ClassNotFoundException e) {
+        LOGGER.debug("Transformation of injectable parameter '{}' failed, the same value of the resolution will be used.",
```

Comment:
why is a class cast something that you can just ignore and continue as is? Shouldn't this actually raise an error to be dealt with?

Reply:
It is a class not found exception. If the current class loader does not have the class being looked for, we will do a best effort and try to inject the value as is to the value provider.